events/ri/README.md

```diff
@@ -0,0 +1,137 @@
+respect to schema handling. The strategy we chose to use here is outlined in [Martin Kleppmann's excellent article] and
+relies on the subject-name strategy called `TopicRecordNameStrategy`. Other strategies are available, see [Robert
+Yokota's follow-up article] for a good overview.
+
+[Martin Kleppmann's excellent article]:https://www.confluent.io/blog/put-several-event-types-kafka-topic/
+[Robert Yokota's follow-up article]:https://www.confluent.io/blog/multiple-event-types-in-the-same-kafka-topic/
+
+### Topic partitioning and message ordering
+
+The `KafkaEventSubscriber` implementation illustrates how to partition the destination Kafka topic _by repository id and
+reference name_: this should allow downstream consumers to process events for a given reference in the order they were
+produced.
+
+Note that the order in which events are produced to a topic does not necessarily reflect the order in which Nessie's
+version store recorded them. As stated in Nessie's [Events API design document], strict ordering of events is not
+guaranteed. More specifically, for two events A and B that happened sequentially in a very short timeframe on the same
+reference, Nessie may notify the subscriber of B first, than A. In this case, the subscriber would put B in the topic
+before A, resulting in out-of-order sequencing.
+
+This only happens for very close events, and for most use cases this shouldn't be a problem; but if strict ordering is
+required, consider using more advanced techniques, such as [Kafka Streams] to re-order the messages, based e.g. on the
+commit timestamp.
+
+[Events API design document]:https://github.com/projectnessie/nessie/blob/nessie-0.60.1/design/events.md?plain=1#L364
+[Kafka Streams]:https://kafka.apache.org/documentation/streams/
+
+### Producer-side filtering
+
+Producer-side filtering is possible by changing the implementation of either the `getEventFilter()` or
+`getEventTypeFilter()` methods.
+
+To illustrate this, `KafkaEventSubscriber` uses its `EventTypeFilter` to completely ignore merge and transplant events.
+(Also note: merge and transplant events also trigger commit events, so it's still possible to capture what happened in a
+branch by listening to commit events only.)
+
+The subscriber is further configured to optionally watch just a few repository ids, or all of them. This is done by
+setting the `nessie.events.repository-ids` configuration option, and is implemented by modifying the subscriber's
+`EventFilter`. Again, this is purely illustrative. Other ways of filtering events on the subscriber side are possible,
+for example one could configure the subscribe to watch only certain branches or tags.
+
+### Using headers efficiently
+
+The `KafkaEventSubscriber` implementation also illustrates how to use Kafka headers to pass additional information along
+with the message. In this case, the subscriber adds the repository id, the user and the event creation timestamp to the
```

Comment:
Did we consider putting `creation timestamp` in the event itself? Instead of headers? It maybe beneficial in the situation when the end user would like to store the events offline for further analysis. If the `creation timestamp` is not there, users will need to create a new avro schema that contains the timestamp and map our event + timestamp to this new schema and finally save the event.

Reply:
Yes, the event timestamp can be part of the message if you prefer. Here I'm using headers, but it's just an example.